### PR TITLE
Jetpack: Improve error handling and clean up feature branch

### PR DESCRIPTION
**Pull request by tonyr59h**
This will be the last PR into the feature branch, after it's in I'll open the PR to get it into `develop`.  #6571 should be merged before starting this. I'm marking it `Not Ready for review` since there's still a couple more changes needed around error handling.

cc @oguzkocer

### CODE REVIEW

**Comment by tonyr59h** [comment]
I decided to remove all the local database stuff pertaining to this feature. None of the settings are used locally and are not editable or viewable without an internet connection so we gain nothing from caching them.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java [hunk 1]
```diff
@@ -318,44 +340,68 @@ public void onResponse(JSONObject response) {
...
                     }
                 });
     }
 
+    private void onFetchResponseReceived(Exception error) {
+        if (error != null) {
+            mWasFetchError = true;
+            mFetchError = error;
+        } else {
+            notifyUpdatedOnUiThread();
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java]: We previously had this call for when an error occurs as well. Any reason we don't need it any more?

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeButtonPrefsFragment.java [hunk 2]
```diff
@@ -197,16 +197,18 @@ public void onErrorResponse(VolleyError error) {
...
 
-        View twitterCard = getView().findViewById(R.id.card_view_twitter);
-        for (int i = 0; i < mPublicizeButtons.size(); i++) {
-            PublicizeButton publicizeButton = mPublicizeButtons.get(i);
-            if (publicizeButton.getId().equals(TWITTER_ID) && publicizeButton.isEnabled()) {
-                twitterCard.setVisibility(View.VISIBLE);
-                return;
+        View view = getView();
+        if (view != null) {
+            View twitterCard = getView().findViewById(R.id.card_view_twitter);
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeButtonPrefsFragment.java]: We can use `view` here instead of `getView()`.

**Comment by tonyr59h** [comment]
> what happens if the activity is re-created, would the data be preserved?

The data would not be preserved in this case, the values would all be set in the fetch response. We already re-fetch the existing site settings when the activity is created, but I can see where this could be a problem. The user could get to Jetpack settings and toggle one of the values before the fetch response and get out of sync.

**Comment by oguzkocer** [comment]
The Jetpack section still _looks_ enabled, but it's not. Is it possible to make it look disabled as well?

screenshot_1504524248

**Comment by oguzkocer** [comment]
I got the following crash while playing with the Settings with "Do not keep activities" enabled and when I put the app to bg. This crash is not necessarily related to this PR, just making a note here and if necessary we can move it to it's own issue/PR.

```
09-04 14:33:56.781 8134-8134/org.wordpress.android E/Parcel: Class not found when unmarshalling: android.support.v7.widget.Toolbar$SavedState
                                                             java.lang.ClassNotFoundException: android.support.v7.widget.Toolbar$SavedState
                                                                 at java.lang.Class.classForName(Native Method)
                                                                 at java.lang.Class.forName(Class.java:400)
                                                                 at android.os.Parcel.readParcelableCreator(Parcel.java:2508)
                                                                 at android.os.Parcel.readParcelable(Parcel.java:2462)
                                                                 at android.os.Parcel.readValue(Parcel.java:2365)
                                                                 at android.os.Parcel.readSparseArrayInternal(Parcel.java:2813)
                                                                 at android.os.Parcel.readSparseArray(Parcel.java:2068)
                                                                 at android.os.Parcel.readValue(Parcel.java:2422)
                                                                 at android.os.Parcel.readArrayMapInternal(Parcel.java:2732)
                                                                 at android.os.BaseBundle.unparcel(BaseBundle.java:269)
                                                                 at android.os.Bundle.getSparseParcelableArray(Bundle.java:934)
                                                                 at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2104)
                                                                 at android.app.Dialog.onRestoreInstanceState(Dialog.java:462)
                                                                 at android.preference.PreferenceScreen.showDialog(PreferenceScreen.java:217)
                                                                 at android.preference.PreferenceScreen.onRestoreInstanceState(PreferenceScreen.java:282)
                                                                 at android.preference.Preference.dispatchRestoreInstanceState(Preference.java:1819)
                                                                 at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:320)
                                                                 at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:325)
                                                                 at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:325)
                                                                 at android.preference.Preference.restoreHierarchyState(Preference.java:1801)
                                                                 at android.preference.PreferenceFragment.onActivityCreated(PreferenceFragment.java:218)
                                                                 at org.wordpress.android.ui.prefs.SiteSettingsFragment.onActivityCreated(SiteSettingsFragment.java:385)
                                                                 at android.app.Fragment.performActivityCreated(Fragment.java:2362)
                                                                 at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1014)
                                                                 at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1171)
                                                                 at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1153)
                                                                 at android.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2039)
                                                                 at android.app.FragmentController.dispatchActivityCreated(FragmentController.java:176)
                                                                 at android.app.Activity.performCreateCommon(Activity.java:6673)
                                                                 at android.app.Activity.performCreate(Activity.java:6681)
                                                                 at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
                                                                 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2618)
                                                                 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
                                                                 at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                                 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
                                                                 at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                 at android.os.Looper.loop(Looper.java:154)
                                                                 at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
                                                              Caused by: java.lang.ClassNotFoundException: android.support.v7.widget.Toolbar$SavedState
                                                                 at java.lang.Class.classForName(Native Method)
                                                                 at java.lang.BootClassLoader.findClass(ClassLoader.java:1346)
                                                                 at java.lang.BootClassLoader.loadClass(ClassLoader.java:1406)
                                                                 at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
                                                                 at java.lang.Class.classForName(Native Method) 
                                                                 at java.lang.Class.forName(Class.java:400) 
                                                                 at android.os.Parcel.readParcelableCreator(Parcel.java:2508) 
                                                                 at android.os.Parcel.readParcelable(Parcel.java:2462) 
                                                                 at android.os.Parcel.readValue(Parcel.java:2365) 
                                                                 at android.os.Parcel.readSparseArrayInternal(Parcel.java:2813) 
                                                                 at android.os.Parcel.readSparseArray(Parcel.java:2068) 
                                                                 at android.os.Parcel.readValue(Parcel.java:2422) 
                                                                 at android.os.Parcel.readArrayMapInternal(Parcel.java:2732) 
                                                                 at android.os.BaseBundle.unparcel(BaseBundle.java:269) 
                                                                 at android.os.Bundle.getSparseParcelableArray(Bundle.java:934) 
                                                                 at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2104) 
                                                                 at android.app.Dialog.onRestoreInstanceState(Dialog.java:462) 
                                                                 at android.preference.PreferenceScreen.showDialog(PreferenceScreen.java:217) 
                                                                 at android.preference.PreferenceScreen.onRestoreInstanceState(PreferenceScreen.java:282) 
                                                                 at android.preference.Preference.dispatchRestoreInstanceState(Preference.java:1819) 
                                                                 at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:320) 
                                                                 at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:325) 
                                                                 at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:325) 
                                                                 at android.preference.Preference.restoreHierarchyState(Preference.java:1801) 
                                                                 at android.preference.PreferenceFragment.onActivityCreated(PreferenceFragment.java:218) 
                                                                 at org.wordpress.android.ui.prefs.SiteSettingsFragment.onActivityCreated(SiteSettingsFragment.java:385) 
                                                                 at android.app.Fragment.performActivityCreated(Fragment.java:2362) 
                                                                 at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1014) 
                                                                 at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1171) 
                                                                 at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1153) 
                                                                 at android.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2039) 
                                                                 at android.app.FragmentController.dispatchActivityCreated(FragmentController.java:176) 
                                                                 at android.app.Activity.performCreateCommon(Activity.java:6673) 
                                                                 at android.app.Activity.performCreate(Activity.java:6681) 
                                                                 at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118) 
                                                                 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2618) 
                                                                 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726) 
                                                                 at android.app.ActivityThread.-wrap12(ActivityThread.java) 
                                                                 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477) 
                                                                 at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                 at android.os.Looper.loop(Looper.java:154) 
                                                                 at android.app.ActivityThread.main(ActivityThread.java:6119) 
                                                                 at java.lang.reflect.Method.invoke(Native Method) 
                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) 
                                                              Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
```

```
81 8134-8134/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
                                                                     Process: org.wordpress.android, PID: 8134
                                                                     java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.prefs.BlogPreferencesActivity}: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.v7.widget.Toolbar$SavedState
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2665)
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
                                                                         at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                         at android.os.Looper.loop(Looper.java:154)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                         at java.lang.reflect.Method.invoke(Native Method)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
                                                                      Caused by: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.v7.widget.Toolbar$SavedState
                                                                         at android.os.Parcel.readParcelableCreator(Parcel.java:2536)
                                                                         at android.os.Parcel.readParcelable(Parcel.java:2462)
                                                                         at android.os.Parcel.readValue(Parcel.java:2365)
                                                                         at android.os.Parcel.readSparseArrayInternal(Parcel.java:2813)
                                                                         at android.os.Parcel.readSparseArray(Parcel.java:2068)
                                                                         at android.os.Parcel.readValue(Parcel.java:2422)
                                                                         at android.os.Parcel.readArrayMapInternal(Parcel.java:2732)
                                                                         at android.os.BaseBundle.unparcel(BaseBundle.java:269)
                                                                         at android.os.Bundle.getSparseParcelableArray(Bundle.java:934)
                                                                         at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2104)
                                                                         at android.app.Dialog.onRestoreInstanceState(Dialog.java:462)
                                                                         at android.preference.PreferenceScreen.showDialog(PreferenceScreen.java:217)
                                                                         at android.preference.PreferenceScreen.onRestoreInstanceState(PreferenceScreen.java:282)
                                                                         at android.preference.Preference.dispatchRestoreInstanceState(Preference.java:1819)
                                                                         at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:320)
                                                                         at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:325)
                                                                         at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:325)
                                                                         at android.preference.Preference.restoreHierarchyState(Preference.java:1801)
                                                                         at android.preference.PreferenceFragment.onActivityCreated(PreferenceFragment.java:218)
                                                                         at org.wordpress.android.ui.prefs.SiteSettingsFragment.onActivityCreated(SiteSettingsFragment.java:385)
                                                                         at android.app.Fragment.performActivityCreated(Fragment.java:2362)
                                                                         at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1014)
                                                                         at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1171)
                                                                         at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1153)
                                                                         at android.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2039)
                                                                         at android.app.FragmentController.dispatchActivityCreated(FragmentController.java:176)
                                                                         at android.app.Activity.performCreateCommon(Activity.java:6673)
                                                                         at android.app.Activity.performCreate(Activity.java:6681)
                                                                         at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2618)
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726) 
                                                                         at android.app.ActivityThread.-wrap12(ActivityThread.java) 
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                         at android.os.Looper.loop(Looper.java:154) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:6119) 
                                                                         at java.lang.reflect.Method.invoke(Native Method) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) 
```

**Comment by tonyr59h** [comment]
> The Jetpack section still looks enabled, but it's not. Is it possible to make it look disabled as well?

Since it's a `PreferenceScreen` I decided to treat just like the `More` preference, allowing users to access the screen while editing is disabled.

**Comment by tonyr59h** [comment]
Those crashes should be resolved as of 7534eba69366b5e7393abeb065a5c9882c9d672c. I also fixed an error restoring state from `BlogPreferencesActivity` in d3b93aefbbebb5b04d41851a24bc43a69db2a23f. This one should be good to go now 👍

**Comment by oguzkocer** [comment]
Looks good! :shipit: 

P.S: Sorry for the long wait on this one! Moving is hard...

**Pull request merged**